src/Eventso.Subscription.Kafka.DeadLetter.Postgres/PoisonEventStore.cs

```diff
@@ -0,0 +1,386 @@
+                    last_failure_timestamp TIMESTAMP    NOT NULL,
+                    last_failure_reason    TEXT         NOT NULL,
+                    total_failure_count    INT          NOT NULL,
+                    PRIMARY KEY (""offset"", partition, topic)
+                );
+
+                CREATE INDEX IF NOT EXISTS ix_poison_events_topic ON eventso_dlq.poison_events (topic);",
+                connection);
+
+            await connection.OpenAsync();
+            
+            await command.ExecuteNonQueryAsync();
+
+            return new PoisonEventStore(connectionFactory);
+        }
+
+        public async Task<long> Count(string topic, CancellationToken cancellationToken)
+        {
+            await using var connection = _connectionFactory.ReadOnly();
+
+            await using var command = new NpgsqlCommand(
+                "SELECT COUNT(*) FROM eventso_dlq.poison_events WHERE topic = @topic;",
+                connection)
+            {
+                Parameters =
+                {
+                    new NpgsqlParameter<string>("topic", topic)
+                }
+            };
+
+            await connection.OpenAsync(cancellationToken);
+
+            var result = await command.ExecuteScalarAsync(cancellationToken);
+            return result != null ? (long)result : 0;
+        }
+
+
+        public async Task<bool> IsStreamStored(string topic, Guid key, CancellationToken token)
+        {
+            await using var connection = _connectionFactory.ReadOnly();
+
+            await using var command = new NpgsqlCommand(
+                "SELECT TRUE FROM eventso_dlq.poison_events WHERE topic = @topic AND key = @key;",
+                connection)
+            {
+                Parameters =
+                {
+                    new NpgsqlParameter<string>("topic", topic),
+                    new NpgsqlParameter<Guid>("key", key)
+                }
+            };
+
+            await connection.OpenAsync(token);
+
+            var result = await command.ExecuteScalarAsync(token);
+            return result != null && (bool)result;
+        }
+
+        public async IAsyncEnumerable<StreamId> GetStoredStreams(
+            IReadOnlyCollection<StreamId> streamIds,
+            [EnumeratorCancellation] CancellationToken token)
+        {
+            await using var connection = _connectionFactory.ReadOnly();
+
+            
+            var topics = new string[streamIds.Count];
+            var keys = new Guid[streamIds.Count];
+            foreach (var (index, streamId) in Enumerate(streamIds))
+            {
+                topics[index] = streamId.Topic;
+                keys[index] = streamId.Key;
+            }
+
+            await using var command = new NpgsqlCommand(
+                @"
+SELECT DISTINCT pe.topic, pe.key
+FROM eventso_dlq.poison_events pe
+INNER JOIN UNNEST(@topics, @keys) AS input(topic, key) 
+ON pe.topic = input.topic AND pe.key = input.key;",
```

Comment:
and here (topic, key) index may be useful

Reply:
just by key will be ok